src/Umbraco.Cms.ManagementApi/Controllers/ExamineManagement/RebuildExamineManagementController.cs

```diff
@@ -0,0 +1,115 @@
+﻿using Examine;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Umbraco.Cms.Infrastructure.Examine;
+using Umbraco.New.Cms.Infrastructure.Services;
+
+namespace Umbraco.Cms.ManagementApi.Controllers.ExamineManagement;
+
+[ApiVersion("1.0")]
+public class RebuildExamineManagementController : ExamineManagementControllerBase
+{
+    private readonly ILogger<RebuildExamineManagementController> _logger;
+    private readonly IIndexRebuilder _indexRebuilder;
+    private readonly ITemporaryIndexingService _temporaryIndexingService;
+    private readonly IExamineManager _examineManager;
+
+    public RebuildExamineManagementController(
+        ILogger<RebuildExamineManagementController> logger,
+        IIndexRebuilder indexRebuilder,
+        ITemporaryIndexingService temporaryIndexingService,
+        IExamineManager examineManager)
+    {
+        _logger = logger;
+        _indexRebuilder = indexRebuilder;
+        _temporaryIndexingService = temporaryIndexingService;
+        _examineManager = examineManager;
+    }
+
+    /// <summary>
+    ///     Rebuilds the index
+    /// </summary>
+    /// <param name="indexName"></param>
+    /// <returns></returns>
+    [HttpPost("rebuild")]
+    [MapToApiVersion("1.0")]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(typeof(OkResult), StatusCodes.Status200OK)]
+    public async Task<IActionResult> Rebuild(string indexName)
+    {
+        if (!_examineManager.TryGetIndex(indexName, out var index))
+        {
+            var invalidModelProblem = new ProblemDetails
+            {
+                Title = "Index Not Found",
+                Detail = $"No index found with name = {indexName}",
+                Status = StatusCodes.Status400BadRequest,
+                Type = "Error",
+            };
+
+            return await Task.FromResult(BadRequest(invalidModelProblem));
+        }
+
+        if (!_indexRebuilder.CanRebuild(index.Name))
+        {
+            var invalidModelProblem = new ProblemDetails
+            {
+                Title = "Could not validate the populator",
+                Detail = $"The index {index?.Name} cannot be rebuilt because we could not validate its associated {typeof(IIndexPopulator)}",
```

Comment:
Nitpicking here, but I would use past tense consequently for the message.

```suggestion
                Detail = $"The index {index?.Name} could not be rebuilt because we could not validate its associated {typeof(IIndexPopulator)}",
```